test/transform/validation_model.jl

```diff
@@ -0,0 +1,81 @@
+
+#
+# Julia implementation of LSTM/RNN classifier for classifying program
+# execution traces. 
+# 
+
+#
+# Loop to check if packages are installed and install them if not, then 
+# import.
+# 
+
+using Pkg
+
+packages = ['Flux','DelimitedFiles','MLDataPattern']
+
+for package in packages
+    haskey(Pkg.installed(),package) || Pkg.add(package)
```

Comment:
You can pull this `Pkg.installed()` out of this loop. Pkg operations are really slow.

Reply:
Sure thing - is there a better way to check that all required packages are installed? Or just let it error out if they aren't?

> I think this is great so far! I added a few small questions in-line.
> 
> Its ok if the demo notebook doesn't allow for generating traces, because of task switching. You can alway run a stand alone command from a notebook by doing
> 
> ```julia
> run(`julia file.jl arg1 arg2`)
> ```
> 
> So you could make the extractor a stand alone script and then call it from the demo notebook. We are doing something a little weird

This was a good idea, but unfortunately it produces the same "task switching" error as just running the `include("output_trace.jl")` approach. The problem seems to be in the trace recursion; the top level trace results are saved out successfully despite the errors, but with that you just get the same information line after line except for the `x` inputs.